### ISSUE TITLE
[RyuJIT/ARM32] pImportSection == pModule->GetImportSectionForRVA(rva) in ExternalMethodFixupWorker

### ISSUE THREAD

**Issue by hqueue** [dotnet/coreclr#12659]
With `COMPlus_AltJit=* COMPlus_AltJitAssertOnNYI=0` options, following assertion is observed.

```
FAILED   - [7080][   9s]readytorun/tests/mainv2/mainv2.sh
               BEGIN EXECUTION
               Microsoft (R) CoreCLR Native Image Generator - Version 4.5.22220.0
               Copyright (c) Microsoft Corporation.  All rights reserved.

               Native image mainv2.ni.exe generated successfully.
               Microsoft (R) CoreCLR Native Image Generator - Version 4.5.22220.0
               Copyright (c) Microsoft Corporation.  All rights reserved.

               Native image fieldgetter.ni.dll generated successfully.
               Microsoft (R) CoreCLR Native Image Generator - Version 4.5.22220.0
               Copyright (c) Microsoft Corporation.  All rights reserved.

               Native image test.ni.dll generated successfully.
               /home/dotnet_jenkins/workspace/hqueue/Run_coreclr_tests_Checked_ryujit_nyi_all/runtime/tests/Tests/coreoverlay/corerun mainv2.exe

               Assert failure(PID 13885 [0x0000363d], Thread: 13885 [0x363d]): pImportSection == pModule->GetImportSectionForRVA(rva)
                   File: /opt/code/github/hqueue/coreclr/src/vm/prestub.cpp Line: 1933
                   Image: /home/dotnet_jenkins/workspace/hqueue/Run_coreclr_tests_Checked_ryujit_nyi_all/runtime/tests/Tests/coreoverlay/corerun

               ./mainv2.sh: line 249: 13885 Aborted                 (core dumped) $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
               Expected: 100
               Actual: 134
               END EXECUTION - FAILED
```

## Failed tests
```
readytorun/tests/mainv1/mainv1.sh
readytorun/tests/mainv2/mainv2.sh
```

**Comment by sjsinju** [comment]
@hqueue @BruceForstall  Please assign this to me.

**Comment by sjsinju** [comment]
With COMPlus_AltJitNgen=* on running tests, it works well.

**Comment by gbalykov** [comment]
I have checked that readytorun.tests.mainv1.mainv1 test passes on 905efa9828b0b02fcbfd339d155982aa4b36f8be and fails on the next commit, which is 9255b5ea1acac68f62207f03dab509de97da93a9 (pull request #12244). Also, readytorun.tests.mainv1.mainv1 test passes on current master with reverted changes of 9255b5ea1acac68f62207f03dab509de97da93a9

**Comment by sjsinju** [comment]
@gbalykov I've checked above tests on current master(https://github.com/dotnet/coreclr/commit/23a09c98aa2cace8616f360ee393837b1701b7f8) with reverted https://github.com/dotnet/coreclr/commit/9255b5ea1acac68f62207f03dab509de97da93a9. but I met following messages
```
               /home/sjsujinkim/works/dotnet/rpi_overlay/overlay/runtime/tests/Tests/coreoverlay/corerun mainv1.exe
               Assert Failure
               Expression: [Recursive resource lookup bug]
               Description: Infinite recursion during resource lookup within System.Private.CoreLib.  This may be a bug in System.Private.CoreLib, or potentially in certain extensibility points such as assembly resolve events or CultureInfo names.  Resource name: IO_EOF_ReadBeyondEOF
               Stack Trace:
                  at System.SR.InternalGetResourceString(String key)
                  at System.SR.GetResourceString(String resourceKey, String defaultString)
                  at System.IO.__Error.EndOfFile()
                  at System.IO.BinaryReader.FillBuffer(Int32 numBytes)
                  at System.IO.BinaryReader.ReadInt32()
                  at System.Resources.ManifestBasedResourceGroveler.CreateResourceSet(Stream store, Assembly assembly)
                  at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists, StackCrawlMark& stackMark)
                  at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo requestedCulture, Boolean createIfNotExists, Boolean tryParents, StackCrawlMark& stackMark)
                  at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
                  at System.Resources.ResourceManager.GetString(String name, CultureInfo culture)
                  at System.SR.InternalGetResourceString(String key)
                  at System.SR.GetResourceString(String resourceKey, String defaultString)
                  at System.IO.__Error.EndOfFile()
                  at System.IO.BinaryReader.FillBuffer(Int32 numBytes)
                  at System.IO.BinaryReader.ReadInt32()
                  at System.Resources.ManifestBasedResourceGroveler.CreateResourceSet(Stream store, Assembly assembly)
                  at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists, StackCrawlMark& stackMark)
                  at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo requestedCulture, Boolean createIfNotExists, Boolean tryParents, StackCrawlMark& stackMark)
                  at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
                  at System.Resources.ResourceManager.GetString(String name, CultureInfo culture)
                  at System.SR.InternalGetResourceString(String key)
                  at System.SR.GetResourceString(String resourceKey, String defaultString)
                  at System.NullReferenceException..ctor()
                  at Program.TestMovedVirtualMethods()
                  at Program.RunAllTests()
                  at Program.Main()
               Expected: 100
               Actual: 35
               END EXECUTION - FAILED
```
Can you check this again?

**Comment by gbalykov** [comment]
@sjsinju

I was unable to reproduce this issue neither on 23a09c9, nor on current master (4b232d817cb2bc1d40a6816407d904581ce1067b). I have recently rebuilt CoreFX and have recently cloned CoreCLR repo from scratch, maybe this could be the case. This issues seem to be related #11318, #12668.

**Comment by hqueue** [comment]
Fixed by #12952
